react/src/components/QrReaderOverlay/QrReaderOverlayContainer.tsx

```diff
@@ -50,127 +44,144 @@ const boxDataToSuccessQrValue = (boxData: BoxDetailsQuery["box"]) => {
-              } as IQrValueWrapper;
-              addQrValueWrapperToMap(resolvedQrValueWrapper);
+              });
             }
+          })
+          .catch((error) => {
+            // eslint-disable-next-line no-console
+            console.error(error);
+            addQrValueWrapperToMap({
+              ...qrValueWrapper,
+              isLoading: false,
+              finalValue: {
+                kind: QrResolverResultKind.FAIL,
+                qrCodeValue: extractedQrCodeFromUrl,
+              },
+            });
           });
-        // TODO: Handle Authorization / No Access To Box case
-        // .catch((error) => {
-        //   console.error(error);
-        //   const resolvedQrValueWrapper = {
-        //     ...qrValueWrapper,
-        //     isLoading: false,
-        //     finalValue: { kind: "noBoxtributeQr" },
-        //   } as IQrValueWrapper;
-        //   return Promise.resolve(resolvedQrValueWrapper);
-        // });
       }
     },
-    [addQrValueWrapperToMap, apolloClient]
+    [addQrValueWrapperToMap, apolloClient],
   );
 
   const handleSingleScan = useCallback(
     (result: string) => {
       const qrCode = extractQrCodeFromUrl(result);
       if (qrCode == null) {
-        console.error("Not a Boxtribute QR Code");
-        onScanningDone([{ kind: "noBoxtributeQr" }]);
+        onScanningDone([{ kind: QrResolverResultKind.NOT_BOXTRIBUTE_QR }]);
       } else {
         apolloClient
-          .query<
-            GetBoxLabelIdentifierForQrCodeQuery,
-            GetBoxLabelIdentifierForQrCodeQueryVariables
-          >({
-            query: GET_BOX_LABEL_IDENTIFIER_BY_QR_CODE,
-            fetchPolicy: "no-cache",
-            variables: { qrCode },
-          })
-          .then(({ data }) => {
-            const boxLabelIdentifier = data?.qrCode?.box?.labelIdentifier;
-            if (boxLabelIdentifier == null) {
-              onScanningDone([
-                { kind: "notAssignedToBox", qrCodeValue: qrCode },
-              ]);
-              console.error("No Box yet assigned to QR Code");
+          .query<GetBoxLabelIdentifierForQrCodeQuery, GetBoxLabelIdentifierForQrCodeQueryVariables>(
+            {
+              query: GET_BOX_LABEL_IDENTIFIER_BY_QR_CODE,
+              fetchPolicy: "no-cache",
+              variables: { qrCode },
+            },
+          )
+          .then(({ data, errors }) => {
+            // eslint-disable-next-line no-console
+            console.log(data, errors);
+
+            if ((errors?.length || 0) > 0) {
+              const errorCode = errors ? errors[0].extensions.code : null;
+              if (errorCode === "FORBIDDEN") {
+                onScanningDone([
+                  { kind: QrResolverResultKind.NOT_AUTHORIZED, qrCodeValue: qrCode },
+                ]);
+              } else if (errorCode === "BAD_USER_INPUT") {
+                onScanningDone([
+                  { kind: QrResolverResultKind.LABEL_NOT_FOUND, qrCodeValue: qrCode },
```

Comment:
In `HeaderMenuContainer.tsx` there is no handling of `LABEL_NOT_FOUND`, is that on purpose?

Reply:
I don't see a way of solving this except for changing the box resolver in the BE.

Reply:
@pylipp Likewise, I think that it makes sense to differentiate between these two cases. However, I can handle it differently when BAD_USER_INPUT returns in the query belong to the QR code, showing a different error case, and vice versa.